drivers/SmartThings/zwave-switch/src/multi-metering-switch/init.lua

```diff
@@ -0,0 +1,140 @@
+  {mfr = 0x015F, prod = 0x3141, model = 0x5102} -- WYFY Touch 4-button Switch
+}
+
+local function can_handle_multi_metering_switch(opts, driver, device, ...)
+  for _, fingerprint in ipairs(MULTI_METERING_SWITCH_FINGERPRINTS) do
+    if device:id_match(fingerprint.mfr, fingerprint.prod, fingerprint.model) then
+      return true
+    end
+  end
+  return false
+end
+
+local function get_profile(device)
+  if device:get_manufacturer() == 0x015F then -- only WYFY's have switch-binary
+    return "switch-binary"
+  else
+    return "metering-switch"
+  end
+end
+
+local function is_zooz_power_strip(device)
+  if device:get_manufacturer() == 0x027A and device:get_model() == 0xA004 then
+    return true
+  end
+end
+
+local function create_child_device(driver, device, children_amount)
+  for i = 2, children_amount+1, 1 do
+    local device_name_without_number = string.sub(driver.label, 0,-2)
+    local name = string.format("%s%d", device_name_without_number, i)
+    local metadata = {
+      type = "EDGE_CHILD",
+      label = name,
+      profile = get_profile(device),
+      parent_device_id = device.id,
+      parent_assigned_child_key = string.format("%02X", i),
+      vendor_provided_label = name,
+    }
+    driver:try_create_device(metadata)
+  end
+end
+
+local function device_added(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZWAVE then
+    local children_amount
+    if is_zooz_power_strip(device) then
+      children_amount = 4;
+    else
+      local endpoints = device.zwave_endpoints
+      children_amount = (#endpoints)-1
+    end
+    create_child_device(driver, device, children_amount)
+  end
+  device:refresh()
+end
+  
+local function find_child(parent, ep_id)
+  if ep_id == PARENT_ENDPOINT then
+    return parent
+  else
+    return parent:get_child_by_parent_assigned_key(string.format("%02X", ep_id))
+  end
+end
+
+local function component_to_endpoint(device, component)
+  return { PARENT_ENDPOINT }
+end
+
+local function device_init(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZWAVE then
+    device:set_find_child(find_child)
+    device:set_component_to_endpoint_fn(component_to_endpoint)
+  end
+end
+
+local function do_refresh(driver, device, command)
+  local component = command and command.component and command.component or "main"
+  if device:is_cc_supported(cc.SWITCH_BINARY) then
+    device:send_to_component(SwitchBinary:Get({}), component)
+  elseif device:is_cc_supported(cc.BASIC) then
+    device:send_to_component(Basic:Get({}), component)
+  end
+  if device:supports_capability_by_id(capabilities.powerMeter.ID) or device:supports_capability_by_id(capabilities.energyMeter.ID) then
+    device:send_to_component(Meter:Get({ scale = Meter.scale.electric_meter.WATTS }), component)
+    device:send_to_component(Meter:Get({ scale = Meter.scale.electric_meter.KILOWATT_HOURS }), component)
+  end
```

Comment:
is this not handled by the default?

Reply:
I would have expected that setting the `component_to_endpoint` function to handle that, no?

Reply:
I assume that something is wrong with default refresh, because as you see I set `component_to_endpoint` function on 88 line. The "Refresh on parent device sends commands to 1 endpoint" test doesn't pass if I delete this overriden refresh function.

Reply:
@cjswedes Can you help me out here? It looks like we might have an error in the z-wave default refresh builder if the parent/main device uses an endpoint rather than no encapsulation

Reply:
Oh. Looks like it's fixed in v46.